os/board/rtl8721csm/src/component/common/file_system/ftl/ftl.c

```diff
@@ -1383,10 +1380,16 @@ uint32_t ftl_write(uint16_t logical_addr, uint32_t w_data)
 
 uint32_t ftl_load_from_storage(void *pdata_tmp, uint16_t offset, uint16_t size)
 {
+	device_mutex_lock(RT_DEV_LOCK_FLASH);
+
 #if defined(CONFIG_AMEBAD_TRUSTZONE) && (FTL_SECURE == 1)
-	return ftl_secure_load_from_storage(pdata_tmp, offset, size);
+	int ret = ftl_secure_load_from_storage(pdata_tmp, offset, size);
+	device_mutex_unlock(RT_DEV_LOCK_FLASH);
+	return ret;
 #else
-	return ftl_non_secure_load_from_storage(pdata_tmp, offset, size);
+	int ret = ftl_non_secure_load_from_storage(pdata_tmp, offset, size);
+	device_mutex_unlock(RT_DEV_LOCK_FLASH);
+	return ret;

```

Comment:
To reduce duplicated lines, how about belows?
```
#if defined()
int ret = ...
#else 
int ret = ...
#endif 
device_mutex_unlock();
return ret;
```

Reply:
I have modified it as suggested